File: Stack.cpp

#include<iostream>
using namespace std;

typedef struct node
{
    int data;
    struct node *next;

}NODE , * PNODE;

class Stack
{
    public:
        PNODE head;
        int iCount;

        Stack()
        {
            head = NULL;
            iCount = 0;
        }

        void Push(int iNo)      //Insertfirst
        { 
            PNODE newn = NULL;

            newn = new NODE;

            newn->data = iNo;
            newn->next = NULL;

            if(head == NULL)
            {
                head = newn;
            }
            else
            {
                newn->next=head;
                head=newn;
            }
            iCount++;
        }


        int Pop()  //delete first
        { 
            PNODE temp = NULL;

            int no=-1;

            if(head == NULL)
            {
                cout<<"Stack is Empty\n";
                return -1;
            }
            else if(head->next == NULL)
            {
                no = head->data;
                delete head;
                head=NULL;
            }
            else
            {
                no=head->data;
                temp = head;
                head = head->next;
                delete temp;
            }
            iCount--;
            return no;
        }

        void Display()
        { 
            
            PNODE temp=head; 

            while(temp !=NULL)
            {
                cout<<temp->data<<"->";
                temp=temp->next;
            }
            cout<<"NULL"<<endl;
        }

        int Count()
        {
            return iCount;
        }
};

int main()
{
  
    int iret=0;

    Stack sobj;
    sobj.Push(101);
    sobj.Push(51);
    sobj.Push(21);
    sobj.Push(11);
   
    cout<<"stack elements are \n";
    sobj.Display();
    iret=sobj.Count();
    
    cout<<"total elements "<<iret<<"\n";

    iret=sobj.Pop();
    cout<<"pop elements is :"<<iret<<"\n";
    iret=sobj.Pop();
    cout<<"pop elements is :"<<iret<<"\n";

    cout<<"stack elements are \n";
    sobj.Display();

    iret=sobj.Count();
    cout<<"total elements "<<iret<<"\n";
    

    return 0;

}

File: Stack.java
class node
{
    public int data;
    public node next;
    
    public node(int X)
    {
        data=X;
        next=null;
    }
}

class Stack
{

    public node head;
    public int icount;

    public Stack()
    {
        head=null;
       icount=0;
    }

    public void Push(int ino)
    {
         node newn=null;

         newn =new node(ino);

         if(head==null)
         {
            head=newn;
         }

         else
         {
            newn.next=head;
            head=newn;
         }
         icount++;
    }

    public void Pop()
    {
        if(head==null)
         {
                 System.out.println("Stack is empty"); 
            return;
         }

         else if(head.next==null)
         {
            head=null;
            System.gc();
         }

         else 
         {
            node temp=head;
            head=head.next;
            System.gc();
         }
        
        icount--;

    }
    public void Display()
    {
        node temp=head;

        while(temp!=null)
        {
            System.out.println(temp.data);
            temp=temp.next; 
        }
    }


    public int Count()
    {
        return icount;
    }

}
public class Mrunal 
{
    public static void main(String Arg[])
    {
         QueueX qobj=new QueueX();

         qobj.Push(51);
        qobj.Push(21);
        qobj.Push(11);
        qobj.Pop();
        qobj.Display();
        int iret=0;
        iret=qobj.Count();

        System.out.println("Elements are :"+iret);
    }
    
}
